proto/akash/deployment/v1beta1/deployment.proto

```diff
@@ -47,6 +47,12 @@ message MsgCreateDeployment {
 
   cosmos.base.v1beta1.Coin deposit = 4
       [(gogoproto.nullable) = false, (gogoproto.jsontag) = "deposit", (gogoproto.moretags) = "yaml:\"deposit\""];
+
```

Comment:
So if this message now includes depositor, it can be a separate wallet address.

When this transaction is broadcast, is there some way to see that `MsgCreateDeployment` is signed by both the deployment owner and the depositor? I'm trying to see how it can be proven that the depositor authorized the deployment to take funds from their wallet.

Reply:
I think it is tracked in the authorizations.  The messages for funding should include the depositor address so we can check the auth.

At least that's how I imagined it.  I haven't read through this yet.

I *believe* that is the pattern that cosmos-sdk is using for allowing gas to be paid by another acct.

Reply:
Yeah! it is handled by the `authz` module through the granted authorizations. We delegate the `MsgSend` to the `authz` module for appropriate processing. See here: https://github.com/ovrclk/akash/pull/1346/files#diff-7892d8c61ce49375a51d974cb679b570f806d3d9fe774f14dbcc8d0ab76ff05bR106-R110